cpp/src/strings/datetime.cu

```diff
@@ -0,0 +1,763 @@
+template<>
+timestamp_units dispatch_timestamp_to_units_fn::operator()<cudf::timestamp_ns>() { return timestamp_units::ns; }
+
+// dispatch operator to map timestamp to native fixed-width-type
+struct dispatch_to_timestamps_fn
+{
+    template <typename T, std::enable_if_t<cudf::is_timestamp<T>()>* = nullptr>
+    void operator()( column_device_view const& d_strings,
+                     std::string const& format,
+                     timestamp_units units,
+                     mutable_column_view& results_view,
+                     cudaStream_t stream ) const
+    {
+        CUDF_EXPECTS( cudf::is_timestamp<T>(), "Unexpected timestamp type" );
+        format_compiler compiler(format.c_str(),units);
+        auto d_items = compiler.compile_to_device();
+        auto d_results = results_view.data<T>();
+        parse_datetime<T> pfn{d_strings,d_items,compiler.items_count(),units};
+        thrust::transform( rmm::exec_policy(stream)->on(stream),
+            thrust::make_counting_iterator<size_type>(0),
+            thrust::make_counting_iterator<size_type>(results_view.size()),
+            d_results, pfn );
+    }
+    template <typename T, std::enable_if_t<not cudf::is_timestamp<T>()>* = nullptr>
+    void operator()( column_device_view const&,
+                     std::string const&, timestamp_units,
+                     mutable_column_view&,
+                     cudaStream_t ) const
+    {
+        CUDF_FAIL("Only timestamps type are expected");
+    }
+};
+
+} // namespace
+
+//
+std::unique_ptr<cudf::column> to_timestamps( strings_column_view const& strings,
+                                             data_type timestamp_type,
+                                             std::string const& format,
+                                             rmm::mr::device_memory_resource* mr = rmm::mr::get_default_resource(),
+                                             cudaStream_t stream = 0 )
+{
+    size_type strings_count = strings.size();
+    if( strings_count==0 )
+        return make_timestamp_column( timestamp_type, 0 );
+
+    CUDF_EXPECTS( !format.empty(), "Format parameter must not be empty.");
+    timestamp_units units = cudf::experimental::type_dispatcher( timestamp_type, dispatch_timestamp_to_units_fn() );
+
+    auto strings_column = column_device_view::create(strings.parent(), stream);
+    auto d_column = *strings_column;
+
+    auto results = make_timestamp_column( timestamp_type, strings_count,
+        copy_bitmask( strings.parent(), stream, mr), strings.null_count(), stream, mr);
+    auto results_view = results->mutable_view();
+    cudf::experimental::type_dispatcher( timestamp_type, dispatch_to_timestamps_fn(),
+                                         d_column, format, units,
+                                         results_view,
+                                         stream );
+    results->set_null_count(strings.null_count());
```

Comment:
Does `strings.null_count()` change between line 405 and here? Otherwise `results` is already constructed with a `null_count()` equal to `strings.null_count()`.

Reply:
Yes it changes on line 406 https://github.com/rapidsai/cudf/blob/d8238cb71d2ab8e6b9d2d8adfc9934a0ca83bd01/cpp/src/strings/datetime.cu#L406
The call to get a `mutable_view()` invalidates the null count. This sets it back to the valid, known value.

Reply:
Ah yes, of course!